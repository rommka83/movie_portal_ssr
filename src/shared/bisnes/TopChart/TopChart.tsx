import React, { FC } from 'react';
import styles from './topchart.module.css';
import { useTranslation } from '../../../i18n';
import { IFilm } from 'shared/types/IFilm';

interface IProps {
  obj: IFilm;
}

export const TopChart: FC<IProps> = ({ obj }) => {
  const { t } = useTranslation();

  return (
    <div className={styles.root}>
      <p className={styles.name}>{t('topChart.Our')}</p>
      <p className={styles.name}>{t('topChart.grade')}</p>

      <input
        type="range"
        className={styles.inp}
        min="0"
        max="10"
        value={obj.rating.kp}
        step="0,1"
        style={{ width: 'width' }}
        id="topChart"
        readOnly
      />
    </div>
  );
};
