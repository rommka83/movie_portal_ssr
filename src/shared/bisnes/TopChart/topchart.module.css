.root {
  display: flex;
  flex-direction: column;
}

.inp:not(:last-child) {
  margin-bottom: 3px;
}

input[type='range'].inp {
  width: 50%;
}

input[type='range'].inp::-webkit-slider-runnable-track {
  height: 10px;
}

.name {
  margin-bottom: 5px;
  color: var(--gray-72);
}
