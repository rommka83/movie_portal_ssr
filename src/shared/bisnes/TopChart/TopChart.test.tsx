import renderer from 'react-test-renderer';
import { TopChart } from './TopChart';

const film = {
  id: 0,
  name: '',
  enName: null,
  year: 0,
  description: null,
  shortDescription: null,
  rating: null,
  movieLength: 0,
  ageRating: null,
  poster: {
    url: '',
    previewUrl: '',
  },
  videos: {
    trailers: [],
    teasers: [],
  },
  genres: [],
  countries: [],
  persons: [],
  similarMovies: null,
  facts: null,
  alternativeName: null,
};

describe('проверка TopChart', () => {
  test('отрисовка TopChart ', () => {
    const component = renderer.create(<TopChart obj={film}></TopChart>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
