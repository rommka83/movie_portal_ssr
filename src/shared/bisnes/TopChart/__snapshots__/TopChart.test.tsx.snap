// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка TopChart отрисовка TopChart  1`] = `
<div
  className="root"
>
  <p
    className="name"
  >
    Our
  </p>
  <p
    className="name"
  >
    grade
  </p>
  <input
    className="inp"
    id="topChart"
    max="10"
    min="0"
    readOnly={true}
    step="0,1"
    style={
      Object {
        "width": "width",
      }
    }
    type="range"
    value={0}
  />
</div>
`;
